plugins/lookup/random_pet.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2021, Abhijeet Kasurde <akasurde@redhat.com>
+# Copyright: (c) 2018, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+    name: random_pet
+    author:
+      - Abhijeet Kasurde (@Akasurde)
+    short_description: Generates random pet names
+    version_added: '3.0.2'
+    requirements:
+      - Python-petname U(https://github.com/dustinkirkland/python-petname).
+    description:
+      - Generates random pet names that can be used as unique identifiers for the resources.
+    options:
+      words:
+        description:
+        - The number of words in the pet name.
+        default: 2
+      length:
+        description: The length of the pet name.
+        default: 6
+      prefix:
+        description: A string to prefix with the name.
+      separator:
+        description: The character to separate words in the pet name.
+        default: "-"
+'''
+
+EXAMPLES = r'''
+- name: Generate pet name
+  ansible.builtin.debug:
+    var: lookup('community.general.random_pet')
+
+- name: Generate pet name with 3 words
+  ansible.builtin.debug:
+    var: lookup('community.general.random_pet', words=3)
+
+- name: Generate pet name with separator
+  ansible.builtin.debug:
+    var: lookup('community.general.random_pet', separator="_")
+
+- name: Generate pet name with length
+  ansible.builtin.debug:
+    var: lookup('community.general.random_pet', length=7)
+'''
+
+RETURN = r'''
+  _raw:
+    description: Random pet name
+    type: str
```

Comment:
```suggestion
    type: list
    elements: str
```
The plugin returns a list of strings (with exactly one string), not a string.